xdsl/transforms/eqsat_add_costs.py

```diff
@@ -0,0 +1,66 @@
+from xdsl.context import MLContext
+from xdsl.dialects import builtin, eqsat
+from xdsl.dialects.builtin import IntAttr
+from xdsl.ir import OpResult
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+
+
+class AddCostEclass(RewritePattern):
+    """
+    Add cost to each operator in an e-class.
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: eqsat.EClassOp, rewriter: PatternRewriter):
+        for operand in op.operands:
+            if not isinstance(operand, OpResult):
+                # only add costs to operators
+                continue
+
+            # Add cost attribute to operators
+            operation = operand.op
+            operation.attributes["cost"] = IntAttr(1)
```

Comment:
this is a pretty common name, and it might clash with another attribute in the future. Let's rename this to `"eqsat_cost"`? Also, could you please extract this to a top-level constant called `EQSAT_COST_LABEL` or similar so that other passes get automatically updated if this is changed in the future?

Reply:
Have moved this upwards in the pass hierarchy but not sure where it should best be defined? In the dialect definition file?